src/api/routes/tx.ts

```diff
@@ -248,6 +251,12 @@ export function createTxRouter(db: DataStore): RouterWithAsync {
       return res.redirect('/extended/v1/tx/0x' + tx_id);
     }
 
+    try {
+      hexToBuffer(tx_id);
+    } catch (error: any) {
+      throw new InvalidRequestError(error.message, InvalidRequestErrorType.invalid_tx_id);
+    }
```

Comment:
Also note if you find quite of a bit places that need this code, you should probably create a helper function in `src/api/query-helpers.ts` like `validateRequestHexInput`